modules/aws_vpc/main.tf

```diff
@@ -12,8 +12,14 @@ locals {
   existing_database_subnets = local.existing_subnets && contains(keys(var.existing_subnet_ids), "database") ? (length(var.existing_subnet_ids["database"]) > 0 ? true : false) : false
 
   #  public_subnets  = local.existing_public_subnets ? data.aws_subnet.public : aws_subnet.public # not used keeping for ref
-  private_subnets = local.existing_private_subnets ? data.aws_subnet.private : aws_subnet.private
+  private_subnets  = local.existing_private_subnets ? data.aws_subnet.private : aws_subnet.private
+  database_subnets = local.existing_database_subnets ? data.aws_subnet.database : aws_subnet.database # tflint-ignore: terraform_unused_declarations
 
+  byon_selector = var.vpc_id == null ? 0 : local.existing_private_subnets ? (var.raw_sec_group_id == null) ? 2 : 3 : 1
```

Comment:
I will create a local for this instead.

Reply:
So it wasn't part of your suggestion, but the "tier" reference is helpful in that it implies that the BYON levels are built upon one another. One thought, we could consider adding "Tier" to the first column heading so that it says "BYON Tier or Scenario"

Reply:
Updated the name to byon_tier, variable was already a local as requested, resolved.